src/vs/workbench/contrib/terminal/common/terminalExtensionPoints.ts

```diff
@@ -27,13 +28,17 @@ export class TerminalContributionService implements ITerminalContributionService
 	private _terminalProfiles: ReadonlyArray<IExtensionTerminalProfile> = [];
 	get terminalProfiles() { return this._terminalProfiles; }
 
+	private _quickFixes: Array<IExtensionTerminalQuickFix> = [];
+	get quickFixes() { return this._quickFixes; }
+
 	constructor() {
 		terminalsExtPoint.setHandler(contributions => {
 			this._terminalProfiles = flatten(contributions.map(c => {
 				return c.value?.profiles?.filter(p => hasValidTerminalIcon(p)).map(e => {
 					return { ...e, extensionIdentifier: c.description.identifier.value };
 				}) || [];
 			}));
+			this._quickFixes = flatten(contributions.map(c => c.value.quickFixes ? c.value.quickFixes.map(fix => { return { ...fix, extensionIdentifier: c.description.identifier.value }; }) : []));
```

Comment:
I'm not sure about the rules of shipping these, does it need to be blocked behind a proposed API check? What about the schema changes?

@mjbvz?

Reply:
Yes this should be gated. Here's an example of this for the static notebook preloads contribution:

https://github.com/microsoft/vscode/blob/182beb6ee4bbed13deabce7ca29f52a30a31824a/src/vs/workbench/contrib/notebook/browser/services/notebookServiceImpl.ts#L500